Add `--force-platform` to `conda-lock install` command.

jezdez opened the pull request:
<!-- Hello! Thanks for submitting a PR! To help make things go a bit more
     smoothly, we would appreciate it if you follow this template. -->

### Description

This selects a specific platform for installation instead of the current, native platform. Should be similar to https://github.com/conda/conda/pull/11794, kinda.

<!-- Good things to put here include:
       - reasons for the change (please link any relevant issues!),
       - any noteworthy (or hacky) choices to be aware of,
       - or what the problem resolved here looked like. -->



<!-- Just as a reminder, everyone in all conda org spaces (including PRs)
     must follow the Conda Org Code of Conduct (link below).

     Finally, once again, thanks for your time and effort. If you have any
     feedback in regards to your experience contributing here, please
     let us know!

     Helpful links:
       - Conda Org COC: https://github.com/conda/.github/blob/main/CODE_OF_CONDUCT.md
       - Contributing docs: https://github.com/conda/.github/blob/main/CONTRIBUTING.md -->

maresb commented:
Thanks @jezdez! This makes sense.

Some initial thoughts:
1. We should add a test
2. I'm debating whether calling it the mundane `--platform` option is a good idea, since the use case is fairly obscure. Probably it's good for consistency, but I think we should at least indicate in the help string that the option is for forcing an override of the native platform, rather than something routine
3. For internal consistency, how about renaming `install(platform=...)` to `install(force_platform=...)`?

jezdez commented:
@maresb Yup, this went to the org admins and I just approved it! (more on the PR tomorrow)

jezdez commented:
> Thanks @jezdez! This makes sense.
> 
> Some initial thoughts:
> 
> 1. We should add a test

I've added one, reusing one of the lockfiles to mimic installing a Intel macOS package on the macOS ARM64 GitHub runners.

> 2. I'm debating whether calling it the mundane `--platform` option is a good idea, since the use case is fairly obscure. Probably it's good for consistency, but I think we should at least indicate in the help string that the option is for forcing an override of the native platform, rather than something routine

I've updated the description

> 3. For internal consistency, how about renaming `install(platform=...)` to `install(force_platform=...)`?

That's fine, I've updated both the argument name as well as the CLI option to make sure it's obvious that it's not quite the same as the `--platform` of the render and lock subcommands.

jezdez commented on tests/test_conda_lock.py:
```diff
@@ -2650,6 +2650,54 @@ def test_fake_conda_env(conda_exe: str, conda_lock_yaml: Path):
+    assert _check_package_installed(
+        package=package,
+        prefix=str(prefix),
+        subdir=platform,
```
It's checking the subdir value of the metadata here

jezdez commented:
Thank you @maresb!

leeleavitt commented:
When will this be released?

maresb commented:
@leeleavitt, apologies for the delay, you can track the progress towards release in https://github.com/conda/conda-lock/issues/725.
